nebula_ros/src/continental/continental_ars548_decoder_ros_wrapper.cpp

```diff
@@ -0,0 +1,344 @@
+    "detection_points", rclcpp::SensorDataQoS());
+  object_pointcloud_pub_ =
+    this->create_publisher<sensor_msgs::msg::PointCloud2>("object_points", rclcpp::SensorDataQoS());
+}
+
+void ContinentalARS548DriverRosWrapper::ReceivePacketsMsgCallback(
+  const nebula_msgs::msg::NebulaPackets::SharedPtr scan_msg)
+{
+  decoder_ptr_->ProcessPackets(*scan_msg);
+}
+
+Status ContinentalARS548DriverRosWrapper::InitializeDriver(
+  std::shared_ptr<drivers::SensorConfigurationBase> sensor_configuration)
+{
+  decoder_ptr_ = std::make_shared<drivers::continental_ars548::ContinentalARS548Decoder>(
+    std::static_pointer_cast<drivers::ContinentalARS548SensorConfiguration>(sensor_configuration));
+
+  decoder_ptr_->RegisterDetectionListCallback(std::bind(
+    &ContinentalARS548DriverRosWrapper::DetectionListCallback, this, std::placeholders::_1));
+  decoder_ptr_->RegisterObjectListCallback(
+    std::bind(&ContinentalARS548DriverRosWrapper::ObjectListCallback, this, std::placeholders::_1));
+
+  return Status::OK;
+}
+
+Status ContinentalARS548DriverRosWrapper::GetStatus()
+{
+  return wrapper_status_;
+}
+
+Status ContinentalARS548DriverRosWrapper::GetParameters(
+  drivers::ContinentalARS548SensorConfiguration & sensor_configuration)
+{
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("host_ip", "255.255.255.255", descriptor);
+    sensor_configuration.host_ip = this->get_parameter("host_ip").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("sensor_ip", "192.168.1.201", descriptor);
+    sensor_configuration.sensor_ip = this->get_parameter("sensor_ip").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("data_port", 2368, descriptor);
+    sensor_configuration.data_port = this->get_parameter("data_port").as_int();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("sensor_model", "");
+    sensor_configuration.sensor_model =
+      nebula::drivers::SensorModelFromString(this->get_parameter("sensor_model").as_string());
+  }
```

Comment:
The default parameters do not reflect the values from the launch file, please adapt them here. 
We should completely get rid of defaults in the code in the near future (#63), so if this does not break anything for the radar, removing the defaults would be even better!

Reply:
Addressed in [31703e5](https://github.com/tier4/nebula/pull/107/commits/31703e5ab2c252069ac509627a2d002f7ba1dd35)

Just in case I will try with the device tomorrow